refactor: enable Mutagen by default on Mac and Windows, fixes #4637 (#4904)
ddev/ddev: docs/content/users/install/performance.md

Mutagen can offer a big performance boost on macOS and Windows. It’s fast and doesn’t need any setup; you only need to enable it. Before Mutagen, Mac and Windows users configured NFS for speed improvements—though it requires setup and isn’t as fast.

Mutagen is enabled by default on Mac and Windows, and it can be disabled per-project or globally.

=== "Mutagen"

    ## Mutagen

    If you’re on macOS or Windows, you can skip everything else here and enable Mutagen:

    ```

>     ddev config global --performance-mode-reset

Author: This is already written above.

Mutagen can offer a big performance boost on macOS and Windows. It’s fast and doesn’t need any setup; you only need to enable it. Before Mutagen, Mac and Windows users configured NFS for speed improvements—though it requires setup and isn’t as fast.

> Mutagen is enabled by default on Mac and Windows but can be disabled globally or also at project level.

Reviewer: Seems like an “and” situation, and I’d reverse the order to subtly suggest handling per-project since there’s presumably good reason for it being globally enabled by default:

```suggestion
Mutagen is enabled by default on Mac and Windows, and it can be disabled per-project or globally.
```

    !!!warning "Don’t Install Mutagen"
        You do not need to install anything to use Mutagen. DDEV installs and maintains its own `mutagen` binary.

>     Mutagen is enabled globally by default. You can disable it with `ddev mutagen reset && ddev config global --performance-mode-reset`.

Author: done

    If you’d rather use Mutagen on a specific project, run [`ddev stop`](../usage/commands.md#stop), enable it with `ddev config --performance-mode=mutagen`, and [`ddev start`](../usage/commands.md#start) again.

    You can’t disable Mutagen on individual projects if it’s enabled globally.

>     To stop using Mutagen on a project, run `ddev mutagen reset && ddev config --performance-mode-reset`.

Author: done

    * **Perform big Git operations on the host side.**<br>
    Git actions that change lots of files, like switching branches, are best done on the host side and not inside the container. You may want to do an explicit `ddev mutagen sync` command after doing something like that to be sure all changes are picked up quickly.
    * **Share projects carefully with non-Mutagen users.**<br>

>     If you share a project with some users that want Mutagen, perhaps on macOS, and other users that don’t want or need it, perhaps on WSL2, don’t check in `.ddev/config.yaml`’s [`performance_mode: "mutagen"`](../configuration/config.md#performance_mode). Instead, either use [global Mutagen configuration](../configuration/config.md#performance_mode) or add a not-checked-in, project-level `.ddev/config.performance_mode.yaml` just to include `performance_mode: "mutagen"` in it. That way, only users with that file will have Mutagen enabled.

Author: done

    ### Troubleshooting Mutagen Sync Issues

>     * Please make sure that DDEV projects work *without* Mutagen before troubleshooting it. Run `ddev config --performance-mode-reset && ddev restart`.

Author: done

    ## NFS

>     !!!warning "NFS is deprecated"

Reviewer 2: I think this is reasonable.

    ## NFS

>     !!!warning "NFS is deprecated"
> 
>         Since users are happy with Mutagen, NFS is now deprecated and should
>         not be longer used anymore. This feature may be removed in an upcoming
>         release.

Reviewer: Shorten and fix line wrapping:

```suggestion
    !!!warning "NFS is deprecated"

        NFS is now deprecated and should no longer be used. This feature may be removed in an upcoming release.
```

I don’t think justification is warranted in this warning—that’s something for a blog post or release notes maybe, worth linking to if it exists. Otherwise best to keep it direct.
